src/webnn_native/nnapi/GraphNnapi.cpp

```diff
@@ -0,0 +1,644 @@
+            biasNode.dimensions = {
+                static_cast<uint32_t>(getOutputChannels(filterNodeInfo.dimensions))};
+            DAWN_TRY(mNnapiMgr->CreateOperandAndSetMemory("bias", &biasNode, &biasMem[0]));
+            biasOpIndex = biasNode.opIndex;
+        } else {
+            biasOpIndex = mGraphNodeMap[conv2d->Inputs()[2].Get()];
+        }
+
+        bool isDepthwiseConv2d = false, isGroupConvolution = false;
+        {
+            if (options->groups > 1) {
+                int32_t inputChannels = 0;
+                if (options->inputLayout == ml::InputOperandLayout::Nchw)
+                    inputChannels = inputNodeInfo.dimensions[1];
+                else if (options->inputLayout == ml::InputOperandLayout::Nhwc)
+                    inputChannels = inputNodeInfo.dimensions[3];
+
+                if (options->groups == inputChannels) {
+                    int32_t filterChannels = 0;
+                    switch (options->filterLayout) {
+                        case ml::FilterOperandLayout::Oihw:
+                        case ml::FilterOperandLayout::Ohwi:
+                            filterChannels = static_cast<int32_t>(filterNodeInfo.dimensions[0]);
+                            break;
+                        case ml::FilterOperandLayout::Hwio:
+                        case ml::FilterOperandLayout::Ihwo:
+                            filterChannels = static_cast<int32_t>(filterNodeInfo.dimensions[3]);
+                            break;
+                        default:
+                            break;
+                    }
+
+                    if (filterChannels == options->groups) {
+                        if (getFilterInChannels(filterNodeInfo.dimensions) == 1) {
+                            isDepthwiseConv2d = true;
+                        } else {
+                            isGroupConvolution = true;
+                        }
+                    }
+                }
+            }
+        }
+
+        int32_t paddingLeft = options->padding ? options->padding[2] : 0;
+        int32_t paddingRight = options->padding ? options->padding[3] : 0;
+        int32_t paddingTop = options->padding ? options->padding[0] : 0;
+        int32_t paddingBottom = options->padding ? options->padding[1] : 0;
+        int32_t strideWidth = options->strides ? options->strides[1] : 0;
+        int32_t strideHeight = options->strides ? options->strides[0] : 0;
+        int32_t dilationsWidth = options->dilations ? options->dilations[1] : 0;
+        int32_t dilationsHeight = options->dilations ? options->dilations[0] : 0;
+        int8_t layout = (options->inputLayout == ml::InputOperandLayout::Nhwc) ? 0 : 1;
+        int32_t groups = options->groups;
+        uint32_t fuseOperation = 0;
+
+        uint32_t paddingLeftOp, paddingRightOp, paddingTopOp, paddingBottomOp, strideWOp, strideHOp;
+        uint32_t fuseOp = 0, layoutOp = 0, dilationWOp = 0, dilationHOp = 0, groupsOp = 0;
```

Comment:
It's better to name as dilationHeightOp, dilationWidthOp, strideWidthOp, strideHeightOp.